lib/src/app.dart

```diff
@@ -1,10 +1,32 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:http/http.dart' as http;
+
+import 'blocs/coins_bloc/coins_bloc.dart';
+import 'blocs/coins_bloc/coins_event.dart';
+import 'screens/home_screen/home_screen.dart';
+import 'services/app_coin_services.dart';
+
 class MyApp extends StatelessWidget {
+  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-      debugShowCheckedModeBanner: false,
-
+    final httpClient = http.Client();
+    return GestureDetector(
+      onTap: () {
+        FocusScopeNode currentFocus = FocusScope.of(context);
+        if (!currentFocus.hasPrimaryFocus &&
+            currentFocus.focusedChild != null) {
+          FocusManager.instance.primaryFocus?.unfocus();
+        }
+      },
```

Comment:
Please let's me know why we have GestureDetector here?

Reply:
We use this to use unfocus feature in the text field when we tap on any point on the screen without a keyboard and search bar. For example in this project, we use it to make the keyboard will turn down and the blinking cursor on the search bar will disappear

Reply:
ok, Mr.Thanh